pkgs/http/lib/src/base_response.dart

```diff
@@ -68,3 +71,34 @@ abstract class BaseResponse {
     }
   }
 }
+
+/// A [BaseResponse] with a [url] field.
+///
+/// [Client] methods that return a [BaseResponse] subclass, such as [Response]
+/// or [StreamedResponse], **may** return a [BaseResponseWithUrl].
+///
+/// For example:
+///
+/// ```dart
+/// final client = Client();
+/// final response = client.get(Uri.https('example.com', '/'));
+/// Uri? finalUri;
+/// if (response case BaseResponseWithUrl(:final url)) {
+///   finalUri = url;
+/// }
+/// // Do something with `finalUri`.
+/// client.close();
+/// ```
+///
+/// [url] will be added to [BaseResponse] when `package:http` version 2 is
+/// released and this mixin will be deprecated.
+mixin BaseResponseWithUrl on BaseResponse {
```

Comment:
I'd make this `abstract interface BaseResponseWithUrl implements BaseResponse`.


Being a mixin with just abstract members means that it is just an interface, you just have to apply it with `with` instead of `implements`.
That's just more complexity for no actual effect. The `abstract interface` is the minimal and idiomatic way to add an abstract member to an interface.

Reply:
That makes sense, thanks! Done.